src/keywrap/mod.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright The ocicrypt Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::config::DecryptConfig;
+use crate::config::EncryptConfig;
+use anyhow::Result;
+use std::collections::HashMap;
+
+/// KeyWrapper is the interface used for wrapping keys using
+/// a specific encryption technology (pgp, jwe, pkcs7, pkcs11, keyprovider)
+pub trait KeyWrapper {
+    /// wrap keys data with encrypt config
+    fn wrap_keys(&self, ec: &EncryptConfig, opts_data: &[u8]) -> Result<Vec<u8>>;
+
+    /// unwrap keys data with decrypt config
+    fn unwrap_keys(&self, dc: &DecryptConfig, annotation: &[u8]) -> Result<Vec<u8>>;
+
+    /// get the keywraper annotation id
+    fn get_annotation_id(&self) -> &str;
```

Comment:
I know we're trying to closely follow the `golang` interface, but typically `Rust` getters will not have the `get_` prefix, so this should be `annotation_id`.
Same applies for the other getters in that interface.